presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxBy.java

```diff
@@ -131,7 +131,7 @@ private InternalAggregationFunction generateAggregation(Type valueType, Type key
                 stateFactory,
                 valueType);
         GenericAccumulatorFactoryBinder factory = AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader);
-        return new InternalAggregationFunction(getSignature().getName(), inputTypes, intermediateType, valueType, true, factory);
```

Comment:
I'm not sure we want to make max_by and min_by order sensitive, either. Those functions are defined as picking the value of x associated with the max value of y, but there's no implication that they are processed in any specific (even if arbitrary) order.

In fact, now that we support ordered aggregations, we should consider deprecating them in favor of an aggregation that returns then first N values.

Reply:
So `array_agg` is the only aggregation function that would use `ORDER BY`?

Reply:
What about `map_agg`?

Reply:
map_agg would use it. It associates a value with the first occurrence of a key, so the order matters.

Eventually, we'll add others such as string_agg.